### PR TITLE
Extract RRD code from app subproject

### PR DESCRIPTION
**Pull request by msnwatson**
## What was the problem?

Now that VRO is pivoting away from using the RRD product, the code specific to the RRD domain needs to be extracted from the `app` Gradle subproject which should focus on code that supports the platform in general

Associated tickets or Slack threads:

https://github.com/department-of-veterans-affairs/abd-vro/issues/1509

## How does this fix it?[^1]
- Extracts code and tests out of the app Gradle project into `domain-rrd`

## How to test this PR
- Run unit tests for `app`, `domain-rrd:rrd-api-controller`, and `domain-rrd:rrd-app-test`
- Ensure that `:dockerComposeUp` still functions
- Run E2E tests that are now located in the `domain-rrd:rrd-app-test` subproject

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: domain-rrd/rrd-app-test/spectral.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+extends: spectral:oas
+rules:
+  operation-description: error
```
Comment (yoomlam) [inline domain-rrd/rrd-app-test/spectral.yaml:3]: When is this used? Do we still need the one in `app`?

**Review by yoomlam — APPROVED** [review]

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4948909756)

File: domain-rrd/rrd-app-test/spectral.yaml (hunk 1 above)
Comment (msnwatson) [reply domain-rrd/rrd-app-test/spectral.yaml]: rm'd

**Comment by msnwatson** [comment]
https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4948909756/jobs/8860710487

**Pull request merged**